### PR TITLE
ENH allow for non-numerical values in objective

### PR DESCRIPTION
**Pull request by tomMoral**
cc @yanislalou, @rflamary

### CODE REVIEW

File: benchopt/stopping_criterion.py [hunk 1]
```diff
@@ -72,7 +72,10 @@ def __init__(self, strategy=None, key_to_monitor='objective_value',
 
         self.kwargs = kwargs
         self.strategy = strategy
-        self.key_to_monitor = key_to_monitor
+        self.key_to_monitor = (
```
Comment (mathurinm) [inline benchopt/stopping_criterion.py:75]: this does another thing that is not related to output processing/saving, do you want it merged in this PR?

File: benchopt/stopping_criterion.py [hunk 2]
```diff
@@ -192,6 +195,15 @@ def should_stop(self, stop_val, objective_list):
             Next value for the stopping criterion. This value depends on the
             sampling strategy for the solver.
         """
+        # Check that the objective is compatible with the stopping_criterion
+        if self.key_to_monitor not in objective_list[0]:
+            key = self.key_to_monitor.replace("objective_", "")
+            raise ValueError(
+                f"Objective output should have a value for '{key}' to be used "
+                "with this stopping_criterion. The name of the key can be "
+                "changed with parameter 'key_to_monitor'."
```
Comment (mathurinm) [inline benchopt/stopping_criterion.py:204]:
```suggestion
                f"Objective.evaluate_result() should contain a key named '{key}' "
                "to be used with this stopping_criterion. The name of the key
                " to use can be changed via the 'key_to_monitor' parameter."
```

File: benchopt/tests/test_objective_outputs.py [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+            objective=objective,
+            solvers=[MINIMAL_SOLVER]
+    ) as benchmark:
+        with pytest.raises(SystemExit, match='1'):
+            with CaptureRunOutput() as out:
+                run([str(benchmark.benchmark_dir),
+                    *'-s test-solver -d test-dataset -n 1 -r 1 --no-display'
+                    .split()], standalone_mode=False)
+
+    out.check_output("Objective output should have a value for 'value'")
```
Comment (mathurinm) [inline benchopt/tests/test_objective_outputs.py:74]:
```suggestion
    out.check_output("Objective.evaluate_result() should contain a key named 'value'")
```

File: benchopt/stopping_criterion.py (hunk 1 above)
Comment (tomMoral) [reply benchopt/stopping_criterion.py:75]: Yes, let's not overcomplicate the process ^^
I usually try to open separate PR but sometimes I take some shortcuts, sorry about that

File: benchopt/stopping_criterion.py [hunk 4]
```diff
@@ -192,6 +195,15 @@ def should_stop(self, stop_val, objective_list):
             Next value for the stopping criterion. This value depends on the
             sampling strategy for the solver.
         """
+        # Check that the objective is compatible with the stopping_criterion
+        if self.key_to_monitor not in objective_list[0]:
+            key = self.key_to_monitor.replace("objective_", "")
+            raise ValueError(
+                f"Objective.evaluate_result() should contain a key named '{key}' "
+                "to be used with this stopping_criterion. The name of the key
+                " to use can be changed via the 'key_to_monitor' parameter."
```
Comment (tomMoral) [inline benchopt/stopping_criterion.py:204]:
```suggestion
                "to be used with this stopping_criterion. The name of the key "
                "can be changed via the 'key_to_monitor' parameter."
```
